### PR TITLE
Upload artifacts from model training

### PR DESCRIPTION
**Pull request by AnnaKwa**
Sometimes there are outputs other than the trained model that we may want to save to the final model directory, such as model checkpoints and training logs. This PR adds a bit to the main training script to upload the contents of the `artifacts` directory to the final dump location. If you want to have additional artifacts saved with the model, configure them to be saved in the `artifacts/` dir during training.

Coverage reports (updated automatically):
- test_unit: [83%](https:\/\/output.circle-artifacts.com\/output\/job\/e53bcf78-c3e3-4445-a0ff-3d5cfe8db49e\/artifacts\/0\/tmp\/coverage\/htmlcov-test_unit\/index.html)

### CODE REVIEW

File: external/fv3fit/docs/usage.rst [hunk 1]
```diff
@@ -17,6 +17,7 @@ It has several useful features:
 
 * Different architectures: RNN, dense, etc...also extensible.
 
+* Any artifacts created during training saved in the `artifacts` directory will be copied to the final output location along with the trained model.
```
Comment (mcgibbon) [inline external/fv3fit/docs/usage.rst:20]: This isn't specific to TransformedModel, is it? Can you put this documentation wherever `fv3fit.train` is documented?

File: external/fv3fit/fv3fit/train.py [hunk 2]
```diff
@@ -177,12 +178,17 @@ def main(args, unknown_args=None):
...
-        logging.basicConfig(
-            level=logging.INFO,
-            format="%(asctime)s [%(levelname)s] %(message)s",
-            handlers=[logging.FileHandler(temp_log.name), logging.StreamHandler()],
-            datefmt="%Y-%m-%d %H:%M:%S",
-        )
-        main(args, unknown_args)
-        copy(temp_log.name, os.path.join(args.output_path, "training.log"))
+    os.makedirs("artifacts", exist_ok=True)
+    logging.basicConfig(
```
Comment (mcgibbon) [inline external/fv3fit/fv3fit/train.py:182]: I like it.

**Review by mcgibbon — APPROVED** [review]
LGTM, though you may want to move the documentation somewhere higher level.

File: external/fv3fit/docs/usage.rst (hunk 1 above)
Comment (AnnaKwa) [reply external/fv3fit/docs/usage.rst:20]: Whoops, saw the doc was named 'usage' and assumed it was general usage. I'll fix this.

**Pull request merged**